src/test/bip32_tests.cpp

```diff
@@ -100,11 +100,69 @@ static void RunTest(const TestVector &test) {
+        BOOST_CHECK(!DecodeExtKey("A" + derive.prv, extkeyTmp));
+        BOOST_CHECK(DecodeExtKey("\n\n\n\n" + derive.prv + "\n\n\n\n", extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey("invalid" + derive.prv, extkeyTmp));
+        BOOST_CHECK(DecodeExtKey(" \r\t\n" + derive.prv + " \n\t\r", extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey(derive.prv + derive.prv, extkeyTmp));
+        BOOST_CHECK(DecodeExtKey("\r" + derive.prv, extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey("\r\a" + derive.prv, extkeyTmp));
+        BOOST_CHECK(DecodeExtKey("\t" + derive.prv, extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey("\a\t" + derive.prv, extkeyTmp));
+        BOOST_CHECK(DecodeExtKey(derive.prv + "\n", extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey(derive.prv + "\a\n", extkeyTmp));
+        BOOST_CHECK(DecodeExtKey(derive.prv + "\r", extkeyTmp));
+        BOOST_CHECK(!DecodeExtKey(derive.prv + "\r\a", extkeyTmp));
 
         // Test public key
         BOOST_CHECK(EncodeExtPubKey(pubkey) == derive.pub);
-        BOOST_CHECK(DecodeExtPubKey(derive.pub) == pubkey); //ensure a base58 decoded pubkey also matches
+        CExtPubKey extpubkey;
+        BOOST_CHECK(DecodeExtPubKey(derive.pub, extpubkey));
+        BOOST_CHECK(extpubkey == pubkey); // ensure a base58 decoded pubkey also matches
+
+        // Test DecodeExtPubKey with some valid and some invalid public keys
+        CExtPubKey extpubkeyTmp;
+        BOOST_CHECK(DecodeExtPubKey(" " + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey("_" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey(derive.pub + " ", extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey(derive.pub + "A", extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey("\n" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey("A" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey("\n\n\n\n" + derive.pub + "\n\n\n\n", extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey("invalid" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey(" \r\t\n" + derive.pub + " \n\t\r", extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey(derive.pub + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey("\r" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey("\r\a" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey("\t" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey("\a\t" + derive.pub, extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey(derive.pub + "\n", extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey(derive.pub + "\a\n", extpubkeyTmp));
+        BOOST_CHECK(DecodeExtPubKey(derive.pub + "\r", extpubkeyTmp));
+        BOOST_CHECK(!DecodeExtPubKey(derive.pub + "\r\a", extpubkeyTmp));
+
+        // Test expected similarities/differences in return values between DecodeExtKey and DecodeExtPubKey
+        BOOST_CHECK(DecodeExtKey(derive.prv, extkeyTmp) != DecodeExtPubKey(derive.prv, extpubkeyTmp));
```

Comment:
I don't understand the style used in this part of the test. It seems like the test would be both more useful and more readable with:

```c++
BOOST_CHECK(DecodeExtKey(derive.prv, extkeyTmp) != DecodeExtPubKey(derive.prv, extpubkeyTmp));
```
replaced by something like:

```c++
BOOST_CHECK_EQUAL(DecodeExtKey(derive.prv, extkeyTmp, true);
BOOST_CHECK_EQUAL(DecodeExtPubKey(derive.prv, extpubkeyTmp), false);
```